enum.rb

```diff
@@ -0,0 +1,186 @@
+module Enumerable
+  def my_each
+    return Enumerator unless block_given?
+
+    ar = self
+    ar.each do |el|
+      yield el
+    end
+  end
+
+  def my_each_with_index
+    return Enumerator unless block_given?
+
+    i = 0
+    ar = self
+    ar.my_each do |element|
+      yield i, element
+      i += 1
+    end
+  end
+
+  def my_select
+    return Enumerator unless block_given?
+
+    ar = self
+    new_arr = []
+    ar.my_each do |i|
+      next unless (yield i) == true
+
+      new_arr.append(i)
+    end
+    new_arr
+  end
+
+  def my_all?(proc = nil)
+    ar = self
+    return ar.all?(proc) if proc
+
+    if block_given?
+      ar.my_each do |i|
+        next unless (yield i) == false
+
+        return false
+      end
+      return true
+    end
+
+    return false if ar.include? false or ar.include? nil
+
+    true
+  end
+
+  def my_none?(proc = nil)
+    ar = self
+    return ar.none?(proc) if proc
+
+    if block_given?
+      ar.my_each do |i|
+        next unless (yield i) == true
+
+        return false
+      end
+    end
+    !ar[ar.length - 1]
+  end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5].my_none?{ |num| num > 5 }  != [1,2,3,4,5].none?{ |num| num > 5 } 
```